### ISSUE TITLE
[Request] Transform method for Cover classes

### ISSUE THREAD

**Issue by lucasimi** [lucasimi/tda-mapper-python#233]
## 🚀 Feature Request

Currently Cover classes expose a `fit` and a `search` method. However, it would be useful to also have a `transform` method that works similarly as `scikit-learn` transformers: you call `fit` on a training set and call `transform` on the test set.

## 🔍 Motivation

The main benefit is that we can construct the Mapper graph on a training set and then check if a test set matches the graph created on the train set.

## 🧠 Algorithm Details (Optional)

The implementation would require that landmarks are somehow stored in the `fit` method, and then used in the `transform` method. The additional landmarks needed to cover the test set should be consistent with the ones already created.
